src/System/Router/Boot.tsx

```diff
@@ -102,34 +105,46 @@ export const Boot = track(undefined, {
       <HeadProvider headTags={headTags}>
         <StateProvider>
           <SystemContextProvider {...contextProps}>
-            <ErrorBoundary>
-              <MediaContextProvider onlyMatch={onlyMatchMediaQueries}>
-                <ResponsiveProvider
-                  mediaQueries={THEME.mediaQueries}
-                  initialMatchingMediaQueries={onlyMatchMediaQueries as any}
-                >
-                  <ToastsProvider>
-                    <StickyProvider>
-                      <AuthIntentProvider>
-                        <AuthDialogProvider>
-                          <DismissibleProvider
-                            userID={props.user?.id}
-                            keys={getProgressiveOnboardingAlertKeys()}
-                          >
-                            <CookieConsentManager>
-                              <FocusVisible />
-                              <SiftContainer />
+            <SystemContextConsumer>
+              {contextValues => {
+                return (
+                  <RelayEnvironmentProvider
+                    environment={contextValues.relayEnvironment}
+                  >
```

Comment:
_This needs some explanation._ 
Initially, I wanted to just inject `RelayEnvironmentProvider` within the `SystemContextProvider`. like this
```
<SystemContextProvider {...contextProps}>
    <RelayEnvironmentProvider environment={contextValues.relayEnvironment}>
    ...
    <RelayEnvironmentProvider />
<SystemContextProvider />
```

However this meant that `contextValues.relayEnvironment` is already defined although it can be `undefined`. The latter gets assigned inside `SystemContextProvider` and is only available when it's ready! This is why I needed to use the `Consumer` here. If you have a more elegant approach, please let me know

Reply:
This works 👍